config.go

```diff
@@ -96,6 +123,11 @@ func DefaultConfig() *Config {
 
 // ValidateConfig is used to validate a sane configuration
 func ValidateConfig(config *Config) error {
+	if config.ProtocolVersion < ProtocolVersionMin ||
+		config.ProtocolVersion > ProtocolVersionMax {
+		return fmt.Errorf("Protocol version %d must be >= %d and <= %d",
+			config.ProtocolVersion, ProtocolVersionMin, ProtocolVersionMax)
+	}
 	if config.HeartbeatTimeout < 5*time.Millisecond {
```

Comment:
Add a new validation check: when ProtocolVersion > 0, LocalID should be nonempty.